src/main/kotlin/interpreter/functions/PyBuiltinFunction.kt

```diff
@@ -69,5 +69,5 @@ abstract class PyBuiltinFunction(val name: String) : PyFunction() {
 
     override var type: PyType
         get() = PyBuiltinFunctionType
-        set(_) = error("Cannot set the type of this object")
+        set(_) = Exceptions.invalidClassSet(this)
```

Comment:
Since `invalidClassSet` is always bound to PyObject, you're better off with adding that func to PyObject itself. Alternatively, add an extension function as util func to reduce overhead for PyObject itself.
```kotlin 
fun <T : PyObject> T.invalidClassSet(): Nothing =
    TYPE_ERROR("Cannot set __class__ on object of type $type.name.").throwKy()
```

Reply:
No. Helper exception methods for pre-built exceptions should be grouped together on Exceptions.